test/integration/cat/test_cat.py

```diff
@@ -49,3 +49,36 @@ def test_cat_xlog_file(tt_cmd, tmp_path):
     rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
     assert rc == 0
     assert re.search(r"replica_id: 1", output)
+
+
+def test_cat_snap_file_timestamp(tt_cmd, tmp_path):
+    # Copy the .snap file to the "run" directory.
+    test_app_path = os.path.join(os.path.dirname(__file__), "test_file", "timestamp.xlog")
+    shutil.copy(test_app_path, tmp_path)
+
+    cmd = [
+        tt_cmd, "cat", "timestamp.xlog", "--timestamp=1731592956.8182"]
+    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
+    assert rc == 0
+    assert re.search(r"lsn: 6", output)
+    assert re.search(r"timestamp: 1731592956.8181", output)
+    assert not re.search(r"lsn: 8", output)
+    assert not re.search(r"timestamp: 1731592956.8184", output)
+
+    cmd = [
+        tt_cmd, "cat", "timestamp.xlog", "--timestamp=2024-11-14T14:02:36.818299999Z"]
+    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
+    assert rc == 0
+    assert re.search(r"lsn: 6", output)
+    assert re.search(r"timestamp: 1731592956.8181", output)
+    assert not re.search(r"lsn: 8", output)
+    assert not re.search(r"timestamp: 1731592956.8184", output)
+
+    cmd = [
+        tt_cmd, "cat", "timestamp.xlog", "--timestamp=2024-11-14T14:02:35Z"]
+    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
+    assert rc == 0
+    assert not re.search(r"lsn: 6", output)
+    assert not re.search(r"timestamp: 1731592956.8181", output)
+    assert not re.search(r"lsn: 8", output)
+    assert not re.search(r"timestamp: 1731592956.8184", output)
```

Comment:
It looks like 3 different tests. Please, split it.

Reply:
Ok. Updated.